apps/designer/app/designer/features/props-panel/props-panel.tsx

```diff
@@ -10,19 +10,99 @@ import type { SelectedInstanceData } from "~/shared/canvas-components";
 import {
   Box,
   Button,
-  Combobox,
-  ComboboxTextField,
-  ComboboxPopperContent,
   Grid,
+  TextField,
   Tooltip,
+  useCombobox,
+  ComboboxPrimitive,
+  ComboboxContent,
+  ComboboxAnchor,
+  ComboboxListbox,
+  ComboboxListboxItem,
+  IconButton,
 } from "@webstudio-is/design-system";
 import {
   PlusIcon,
   TrashIcon,
   ExclamationTriangleIcon,
+  ChevronDownIcon,
 } from "@webstudio-is/icons";
 import { handleChangePropType, usePropsLogic } from "./use-props-logic";
 
+type ComboboxProps = {
+  isReadonly: boolean;
+  isInvalid: boolean;
+  items: Array<string>;
+  value: string;
+  onItemSelect: (value: string | null) => void;
+  onSubmit: (value: string) => void;
+};
+
+export const Combobox = ({
+  isReadonly,
+  isInvalid,
+  items: itemsProp,
+  value,
+  onItemSelect,
+  onSubmit,
+}: ComboboxProps) => {
+  const {
+    items,
+    getInputProps,
+    getComboboxProps,
+    getToggleButtonProps,
+    getMenuProps,
+    getItemProps,
+    isOpen,
+  } = useCombobox({
+    items: itemsProp,
+    value,
+    itemToString: (item) => item ?? "",
+    onItemSelect,
+  });
+
+  return (
+    <ComboboxPrimitive>
+      <Box {...getComboboxProps()}>
+        <ComboboxAnchor>
+          <TextField
+            {...getInputProps()}
+            name="prop"
+            placeholder="Property"
+            readOnly={isReadonly}
+            state={isInvalid ? "invalid" : undefined}
+            suffix={
+              <IconButton {...getToggleButtonProps()}>
+                <ChevronDownIcon />
+              </IconButton>
+            }
+            onKeyPress={(event) => {
```

Comment:
I’d move this to props getter to be safe you don’t override what’s inside.

Reply:
done